lib/mixlib/install/backend.rb

```diff
@@ -0,0 +1,51 @@
+#
+# Author:: Patrick Wright (<patrick@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'net/http'
+require 'json'
+
+module Mixlib
+  class Install
+    class Backend
+      def self.info(options)
+        parameters = {
```

Comment:
Here I would do something like:

```
backend = if ['current', 'stable'].include? options.channel
  Backend::Omnitruck.new(options)
end

backend.info
```

And move the rest of the logic into Omnitruck class.